### PR TITLE
feat: strf-9440 Stencil Bundle: fail on scss failure compilation

### PR DESCRIPTION
**Pull request by jairo-bc**
#### What?

Scss compilation validation task during theme bundling by running `stencil bundle` command 

#### Tickets / Documentation

-   [STRF-9440](https://jira.bigcommerce.com/browse/STRF-9440)

#### Screenshots (if appropriate)
Screenshot 2022-03-09 at 20 19 02

Screenshot 2022-03-09 at 20 47 32
cc @bigcommerce/storefront-team

### CODE REVIEW

File: server/plugins/theme-assets/theme-assets.module.js [hunk 1]
```diff
@@ -62,35 +57,10 @@ internals.cssHandler = async (request, h) => {
...
-            browsers: configuration.autoprefixer_browsers,
-        },
-    };
-    const stencilStyles = new StencilStyles(console);
-    stencilStyles.activateEngine(SASS_ENGINE_NAME);
+    const themeAssetsPath = internals.getThemeAssetsPath();
 
     try {
-        const css = await stencilStyles.compileCss('scss', params);
+        const css = cssCompiler.compile(configuration, themeAssetsPath, fileName);
```
Comment (jmwiese) [inline server/plugins/theme-assets/theme-assets.module.js:63]: Do we need to `await cssCompiler.compile` here?

Comment (jairo-bc) [reply server/plugins/theme-assets/theme-assets.module.js]: yes!

**Review by jmwiese — APPROVED** [review]

File: lib/bundle-validator.js [hunk 2]
```diff
@@ -402,6 +404,42 @@ class BundleValidator {
...
+        try {
+            const files = await fs.promises.readdir(stylesPath);
+            cssFiles = files.filter((file) => {
+                return file.slice(-(compiler.length + 1)) === `.${compiler}`;
+            });
+        } catch (e) {
+            throw new Error(`Error: ${e.message}, while reading files from "${stylesPath}".`.red);
+        }
+
+        for await (const file of cssFiles) {
```
Comment (jmwiese) [inline lib/bundle-validator.js:423]: Does the use of `for await` here mean that `cssFiles` is an  async interable object? It seems like we're doing `await` inside of the `try` so dose that mean the filter/slice operation results in `async`?

Comment (jairo-bc) [reply lib/bundle-validator.js]: I think it was a leftover from previous solution. But you are right, that `for await` should be used for async iterable objects. On other hand it also supports `sync` iterables, I think it's cleaner to separate those concerns

Comment (jairo-bc) [reply lib/bundle-validator.js]: I think the problem was in the eslint as well: https://eslint.org/docs/rules/no-await-in-loop
But those operation steps should go in sync way, so I disabled that rule for the next line

**Comment by mattcoy-arcticleaf** [comment]
@jairo-bc , sorry, this appears to now be fixed in 4.1!

**Pull request merged**